src/compute/src/sink/persist_sink.rs

```diff
@@ -426,6 +413,27 @@ where
         // do this, we would be stuck at `[minimum]`.
         let mut emitted_persist_frontier: Option<Antichain<_>> = None;
 
+        // We go straight to the write handle to learn about the current upper
+        // and upper advancements. The "persist_oks" stream is lying to us! For
+        // example, when starting the persist_source with an as_of, the upper
+        // will jump to that as_of.
+        let mut current_upper = persist_frontier.clone();
+        let upper_stream = async_stream::stream!({
+            loop {
+                write.wait_for_upper_past(&current_upper).await;
```

Comment:
Is this call cancellation-safe?

Reply:
I don't know, but the way we consume from the stream is cancel safe. I'll add a note to where we call `next()`, about cancel safety!